Developer controls

getmikyled opened the pull request:
- Creates an enemies prefab folder to access from for 'spawn' command.
- Adds the developer console script

Contains the commands:
- help
- killable
- kill
- restart
- spawn

getmikyled committed:
Add Script

getmikyled committed:
Update DeveloperControls.cs

getmikyled committed:
Updated the DeveloperControls Script

Now has the ability to input the killable command.
Command does not have functionality yet

getmikyled committed:
Implemented SpawnEnemy(enemy) & Restart()

Restart() reloads the scene
SpawnEnemy spawns an enemy

getmikyled committed:
implemented more functionality to developer console

getmikyled committed:
Fixed spawn enemy command

getmikyled committed:
Implemented 3 New Commands

Adds
- Killable Command
- Kill Command
- Help Command

getmikyled committed:
Changed Tool Name

getmikyled committed:
Fixed Variables

getmikyled committed:
Added exception and changed formatting

TheStress commented on Assets/Scripts/Tools/Editor/DeveloperControls.cs:
```diff
@@ -0,0 +1,242 @@
+
+    private void OnEnable()
+    {
+        player = GameObject.Find("Player"); // finds the player's game object for modification
```
I would recommend extracting this finding player code and turning it into its own function because you use it multiple times in this script. Also I would change how you find the player object because using its object name can be volitile, instead maybe try using tags instead.

TheStress commented on Assets/Scripts/Tools/Editor/DeveloperControls.cs:
```diff
@@ -0,0 +1,242 @@
+
+        if (state) 
+        {
+            player.GetComponent<PlayerKillable>().isKillable = true; // sets killable to true
```
Make sure you pull from the develop branch and test that your code works with the existing code

TheStress reviewed (commented):
Looks good but I wanna make sure that everything still works first before approving because you havent tested it with the current version of the develop branch.

getmikyled committed:
Merge branch 'develop' into developerControls

getmikyled committed:
Updated Developer Consoles and Player Prefab

 - Changed the tag of the player prefab to be player in order to work with the developer console tool.
- Fixed an issue that prevented any input in the game view when the developer console is open.
- Cleaned up some code

TheStress committed:
Update DeveloperControls.cs

getmikyled committed:
Update DeveloperControls.cs

Fixed some code to ensure that the console log is displayed

getmikyled committed:
Merge branch 'develop' into developerControls

getmikyled committed:
Moved Folders & Began Developing new Developer Console

getmikyled committed:
Created Dev Console Prefab & Creates Console on Scene Load

getmikyled committed:
Fixed initilization and cleaned up code

TheStress commented on Assets/Tools/DeveloperConsole.cs:
```diff
@@ -0,0 +1,273 @@
+    // an array that contains all the possible commands
+    string[] commandArray = new string[]
+    {
+        "help", "killable", "kill", "restart", "spawn"
```
I dont know if you are still in the middle of developing it, but there is no change **scene command**. Also, I added another command to the wiki, its just **listing out all scenes alphabetically** just so people can know what the names of the scenes are

getmikyled replied:
Sorry, I have been busy during the break so I haven't had much time to work on it. I will try to have it done soon though!

TheStress replied:
You don't need to apologize, work is optional during the break. I was just leaving comments cause I saw that you were working on this and I thought you were done. You can get to it whenever you want.

getmikyled committed:
Update Developer Console Design

getmikyled committed:
Added Commands

- load scene
- scene_list

getmikyled committed:
Fix DevConsoleUI & Remove Editor Window Version

getmikyled commented:
Note: Also moved tools folder to be separate from scripts, so that is why it'll show more changes

getmikyled committed:
Move DevConsole Prefab into Resources/Tools

getmikyled committed:
Fix Tool Initialization to not use UnityEditor

getmikyled commented:
Finished making changes!

TheStress commented on Assets/Tools/DeveloperConsole.cs:
```diff
@@ -0,0 +1,389 @@
+            //---------------------
+            // EXECUTE THE COMMANDS
+            //---------------------
+            if (commandType == DevCommand.help) // checks for help command which shows documentation
```
Can you use `switch` statement instead of all the `if() else if()` for checking `commandType` enum?

getmikyled replied:
Fixed!

TheStress reviewed (approved):
Looks good besides a small change merge when your good with it

getmikyled committed:
Changed if-elseif to switch case for commands
